### PR TITLE
Process oEmbed at custom query

### LINKED ISSUE #23: Plain link and/or prepended with "oembed:" option
Indicate the use of plain link: https://www.youtube.com/embed/2Xc9gXyf2G4 or this style oembed: https://www.youtube.com/embed/2Xc9gXyf2G4 or both in options.

### PR DESCRIPTION
**Pull request by rayriffy**
close #23

### CODE REVIEW

**Comment by raae** [comment]
Thank you for submitting this. I would there to be a config option changing the behavior so the plugins stay backward compatible. 

Could you add that?

Also, I would like to keep selecting nodes first and then processing them. It makes testing easier.

**Comment by rayriffy** [comment]
Here's an update

- I adding options `newFilter` to the code which will use new custom query when if set to `true`
- `newFilter` will be `false` by default when it's undefined
- I managed to use `selectPossibleOembedLinkNodes` with few possible changes
- An extra parameters in `selectPossibleOembedLinkNodes` will be return  `false` by default to prevent old test cases from `jest` to fail

PS. It is possible to fix #40, too by passing `debug: true` to enable logging

File: helpers.js [hunk 1]
```diff
@@ -171,8 +171,11 @@ exports.fetchOembed = async endpoint => {
   return response.data;
 };
 
-exports.selectPossibleOembedLinkNodes = markdownAST => {
-  return select(markdownAST, "paragraph link:only-child");
+exports.selectPossibleOembedLinkNodes = (markdownAST, newFilter = false) => {
+  if (newFilter)
+    return select(markdownAST, "inlineCode");
```
Comment (raae) [inline helpers.js]:
```suggestion
return select(markdownAST, "inlineCode").
 map(node => {
        if (!node.value.startsWith("oembed:")) return;
        node.url = node.value.substring(7);
      });
```

**Comment by raae** [comment]
Yes, #40 could be solved by a debug setting. But I would like to use the official Gatsby way, see #15. I hope this lets people use global setting...

**Comment by raae** [comment]
Also, send me a DM on Twitter or an e-mail (raae@raae.codes) with your postal address and I will hook you up with some Norwegian chocolate as thanks.

File: index.js [hunk 2]
```diff
@@ -7,31 +7,31 @@ const {
...
-const processNode = async (node, providers) => {
+const processNode = async (node, providers, reporter) => {
   try {
-    console.log(`Process node ${node.url}`);
+    reporter.info(`Process node ${node.url}`);
     // Check if url matched any of the oembed url schemes.
-    const endpoint = getProviderEndpointForLinkUrl(node.url, providers);
-    console.log(`With oembed request ${JSON.stringify(endpoint)}`);
+    const endpoint = getProviderEndpointForLinkUrl(node.url, providers, reporter);
+    reporter.info(`With oembed request ${JSON.stringify(endpoint)}`);
```
Comment (rayriffy) [inline index.js]: Is this require for logging? Because if this log removed, console will a lot cleaner

File: helpers.js [hunk 3]
```diff
@@ -171,8 +171,20 @@ exports.fetchOembed = async endpoint => {
   return response.data;
 };
 
-exports.selectPossibleOembedLinkNodes = markdownAST => {
-  return select(markdownAST, "paragraph link:only-child");
+exports.selectPossibleOembedLinkNodes = (markdownAST, usePrefix = false) => {
+  if (usePrefix === true) {
+    const nodes = select(markdownAST, "inlineCode");
+    var res = [];
+    nodes.map(node => {
```
Comment (rayriffy) [inline helpers.js]: This line will get all nodes with `{type: 'inlineCode'}`. Then, I filltering it and then push to `res` array

File: index.js (hunk 2 above)
Comment (raae) [reply index.js]: Let's keep it in. I will do a clean up after merging this PR.

**Comment by raae** [comment]
Before releasing to npm I will:
* create tests
* clean up logging and docs 

Thank you so much @rayriffy :D

**Pull request merged**